src/main/java/com/puppycrawl/tools/checkstyle/utils/CommonUtil.java

```diff
@@ -369,6 +369,8 @@ public static URI getUriByFilename(String filename) throws CheckstyleException {
      * @param filename name of the file
      * @return resolved file URI or null if URL is malformed or non-existent
      */
+    // disabled until https://github.com/checkstyle/checkstyle/issues/17646
+    @SuppressWarnings("deprecation")
```

Comment:
Please use noinspection javadoc annotations, see examples around our codebase.
There is one more javadoc annotation is required to provide a reason.

Java annotation that you used will suppress all deprecation even from other tools